libcrux-psq/src/handshake/ciphersuite.rs

```diff
@@ -0,0 +1,747 @@
+            Option<<Self as CiphersuiteBase>::SharedSecret>,
+        ),
+        HandshakeError,
+    >;
+}
+pub trait ResponderCiphersuiteTrait: CiphersuiteBase {
+    type DecapsulationKey;
+    fn own_pq_encapsulation_key(&self) -> Option<<Self as CiphersuiteBase>::EncapsulationKeyRef>;
+
+    fn pq_decapsulate(
+        &self,
+        ciphertext: Option<&<Self as CiphersuiteBase>::Ciphertext>,
+    ) -> Result<Option<<Self as CiphersuiteBase>::SharedSecret>, HandshakeError>;
+}
+
+pub struct ResponderX25519ChaChaPolyHkdfSha256<'a> {
+    pub longterm_ecdh_keys: &'a DHKeyPair,
+}
+
+pub struct ResponderX25519MlKem768ChaChaPolyHkdfSha256<'a> {
+    pub longterm_ecdh_keys: &'a DHKeyPair,
+    pub longterm_pq_encapsulation_key: &'a MlKem768PublicKey,
+    pub longterm_pq_decapsulation_key: &'a MlKem768PrivateKey,
+}
+pub struct InitiatorX25519Mlkem768ChachaPolyHkdfSha256<'a> {
+    pub longterm_ecdh_keys: &'a DHKeyPair,
+    pub peer_longterm_ecdh_pk: &'a DHPublicKey,
+    pub peer_longterm_mlkem_pk: &'a libcrux_ml_kem::mlkem768::MlKem768PublicKey,
+}
+pub struct InitiatorX25519ChachaPolyHkdfSha256<'a> {
+    pub longterm_ecdh_keys: &'a DHKeyPair,
+    pub peer_longterm_ecdh_pk: &'a DHPublicKey,
+}
+
+#[cfg(feature = "classic-mceliece")]
+pub struct InitiatorX25519ClassicMcElieceChachaPolyHkdfSha256<'a> {
+    pub longterm_ecdh_keys: &'a DHKeyPair,
+    pub peer_longterm_ecdh_pk: &'a DHPublicKey,
+    pub peer_longterm_cmc_pk: &'a PublicKey,
+}
+#[cfg(not(feature = "classic-mceliece"))]
+pub struct InitiatorX25519ClassicMcElieceChachaPolyHkdfSha256 {}
+#[cfg(feature = "classic-mceliece")]
+pub struct ResponderX25519ClassicMcElieceChaChaPolyHkdfSha256<'a> {
+    pub longterm_ecdh_keys: &'a DHKeyPair,
+    pub longterm_pq_encapsulation_key: &'a PublicKey,
+    pub longterm_pq_decapsulation_key: &'a SecretKey,
+}
+
+impl<'a> ResponderCiphersuiteTrait for ResponderCiphersuite<'a> {
+    type DecapsulationKey = DynamicDecapsulationKey;
+
+    fn own_pq_encapsulation_key(&self) -> Option<<Self as CiphersuiteBase>::EncapsulationKeyRef> {
+        match self {
+            ResponderCiphersuite::X25519_MlKem768_ChaChaPoly_HkdfSha256(
+                responder_x25519_ml_kem768_cha_cha_poly_hkdf_sha256,
+            ) => Some(DynamicEncapsulationKeyRef::MlKem(
+                responder_x25519_ml_kem768_cha_cha_poly_hkdf_sha256.longterm_pq_encapsulation_key,
+            )),
+            #[cfg(feature = "classic-mceliece")]
+            ResponderCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(
+                responder_x25519_cmc_cha_cha_poly_hkdf_sha256,
+            ) => Some(DynamicEncapsulationKeyRef::CMC(
+                responder_x25519_cmc_cha_cha_poly_hkdf_sha256.longterm_pq_encapsulation_key,
+            )),
+            #[cfg(not(feature = "classic-mceliece"))]
+            ResponderCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(_) => {
+                panic!("unsupported ciphersuite")
```

Comment:
Maybe let's return `None` in these cases and print an error + `debug_assert`. Just to avoid panics at runtime in deployments.